streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyWindowStore.java

```diff
@@ -89,9 +128,28 @@
      * @return an iterator over windowed key-value pairs {@code <Windowed<K>, value>}
      * @throws InvalidStateStoreException if the store is not initialized
      * @throws NullPointerException If {@code null} is used for any key.
+     * @deprecated Use {@link #fetch(Object, Object, Instant, Duration)} instead
      */
+    @Deprecated
     KeyValueIterator<Windowed<K>, V> fetch(K from, K to, long timeFrom, long timeTo);
-    
+
+    /**
+     * Get all the key-value pairs in the given key range and time range from all the existing windows.
+     * <p>
+     * This iterator must be closed after use.
+     *
+     * @param from      the first key in the range
+     * @param to        the last key in the range
+     * @param fromTime  time range start (inclusive)
+     * @param duration  time range duration
+     * @return an iterator over windowed key-value pairs {@code <Windowed<K>, value>}
+     * @throws InvalidStateStoreException if the store is not initialized
+     * @throws NullPointerException If {@code null} is used for any key.
+     * @throws IllegalArgumentException if duration is negative or can't be represented as {@code long milliseconds}
+     */
+    KeyValueIterator<Windowed<K>, V> fetch(K from, K to, Instant fromTime, Duration duration)
```

Comment:
I just talked to @guozhangwang @bbejeck and @vvcephei -- it seems, we are in favor to go with `Instant/Instant` types here and also for `fetchAll()`. `2.1` branch is not cut, and thus we can still get this into 2.1 -- can you update the PR accordingly? (Sorry for the long delay, but we were quite busy with many other deadline related things...)

Reply:
@mjsax 

I've updated (KIP-358)[https://cwiki.apache.org/confluence/display/KAFKA/KIP-358%3A+Migrate+Streams+API+to+Duration+instead+of+long+ms+times] and PR according to your proposal.

Please, review.